drivers/counter/ds3231.c

```diff
@@ -0,0 +1,1063 @@
+		 * to also record the tm_mday for our selected base
+		 * 1978-01 that will produce the correct tm_wday.
+		 */
+		tm.tm_mday = (*dp & 0x07);
+		tm.tm_wday = tm.tm_mday - 1;
+	} else {
+		tm.tm_mday = 10 * ((*dp >> 4) & 0x3) + (*dp & 0x0F);
+	}
+
+	/* Walk backwards to extract the alarm mask flags. */
+	while (true) {
+		if (REG_ALARM_IGN & *dp) {
+			flags |= amf;
+		}
+		amf >>= 1;
+		if (dp-- == ap) {
+			break;
+		}
+	}
+
+	/* Convert to the reduced representation. */
+	*tp = timeutil_timegm(&tm);
+	return flags;
+}
+
+static int encode_alarm(u8_t *ap,
+			bool with_sec,
+			time_t time,
+			u8_t flags)
+{
+	struct tm tm;
+	u8_t val;
+
+	(void)gmtime_r(&time, &tm);
+
+	/* For predictable behavior the low 4 bits of flags
+	 * (corresponding to AxMy) must be 0b1111, 0b1110, 0b1100,
+	 * 0b1000, or 0b0000.  This corresponds to the bitwise inverse
+	 * being one less than a power of two.
+	 */
+	if (!is_power_of_two(1U + (0x0F & ~flags))) {
+		LOG_DBG("invalid alarm mask in flags: %02x", flags);
+		return -EINVAL;
+	}
+
+	if (with_sec) {
+		if (flags & RTC_DS3231_ALARM_FLAGS_IGNSE) {
+			val = REG_ALARM_IGN;
+		} else {
+			val = ((tm.tm_sec / 10) << 4) | (tm.tm_sec % 10);
+		}
+		*ap++ = val;
+	}
+
+	if (flags & RTC_DS3231_ALARM_FLAGS_IGNMN) {
+		val = REG_ALARM_IGN;
+	} else {
+		val = ((tm.tm_min / 10) << 4) | (tm.tm_min % 10);
+	}
+	*ap++ = val;
+
+	if (flags & RTC_DS3231_ALARM_FLAGS_IGNHR) {
+		val = REG_ALARM_IGN;
+	} else {
+		val = ((tm.tm_hour / 10) << 4) | (tm.tm_hour % 10);
+	}
+	*ap++ = val;
+
+	if (flags & RTC_DS3231_ALARM_FLAGS_IGNDA) {
+		val = REG_ALARM_IGN;
+	} else if (flags & RTC_DS3231_ALARM_FLAGS_DOW) {
+		val = REG_DAYDATE_DOW | (tm.tm_wday + 1);
+	} else {
+		val = ((tm.tm_mday / 10) << 4) | (tm.tm_mday % 10);
+	}
+	*ap++ = val;
+
+	return 0;
+}
+
+static u32_t decode_rtc(struct ds3231_data *data)
+{
+	struct tm tm = { 0 };
+	const struct register_map *rp = &data->registers;
+
+	decode_time(&tm, &rp->sec, true);
+	tm.tm_wday = (rp->dow & 0x07) - 1;
+	tm.tm_mday = 10 * ((rp->dom >> 4) & 0x03) + (rp->dom & 0x0F);
+	tm.tm_mon = 10 * (((0xF0 & ~REG_MONCEN_CENTURY) & rp->moncen) >> 4)
+		    + (rp->moncen & 0x0F) - 1;
+	tm.tm_year = (10 * (rp->year >> 4)) + (rp->year & 0x0F);
+	if (REG_MONCEN_CENTURY & rp->moncen) {
+		tm.tm_year += 100;
+	}
+
+	data->rtc_registers = timeutil_timegm(&tm);
+	return data->rtc_registers;
+}
+
+static int update_registers(struct device *dev)
+{
+	struct ds3231_data *data = dev->driver_data;
+	const struct ds3231_config *cfg = dev->config->config_info;
+	u32_t syncclock;
+	int rc;
+	u8_t addr = 0;
+
+	data->syncclock_base = rtc_ds3231_read_syncclock(dev);
+	rc = i2c_write_read(data->i2c, cfg->addr,
+			    &addr, sizeof(addr),
+			    &data->registers, sizeof(data->registers));
+	syncclock = rtc_ds3231_read_syncclock(dev);
+	if (rc < 0) {
+		return rc;
+	}
+	data->rtc_base = decode_rtc(data);
+
+	printk("Time %u at %u (%d)\n", (u32_t)data->rtc_base,
```

Comment:
This looks like some debug code left behind.

Reply:
removed

Reply:
is it?

Reply:
Was probably lost during the restoration when I deleted the branch and github wouldn't allow the PR to be reopened.  Gone now.